### PR TITLE
Support Helm installation in SMI extension

### PR DESCRIPTION
**Pull request by Pothulapati**
Support for Helm is (kind-off) present by default already, because we use
the same templates for the CLI binary too. 

This PR adds the following:
- Add Helper scripts to make sure Helm Readme is auto-generated and updated.
- Add an Integration test which tests the SMI Extension Helm chart.

Now, The integration test is separated into two parts i.e Helm, CLI. 
In the Helm case, Linkerd2 core control-plane is also installed through
Helm.

As these tests are present in the same integration tests file, these
are parallelized by using workflow strategies and creating separate
kind clusters for each, and running them independently.

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -90,4 +97,9 @@ jobs:
         curl -sL https://run.linkerd.io/install-edge | sh
     - name: Run Integration Tests
       run: |
-        GO111MODULE=on go test --failfast --mod=readonly "./test" --linkerd="$HOME/.linkerd2/bin/linkerd" --integration-tests
+        if [ '${{ matrix.test }}' == 'helm' ]; then
+          projectdir=$(pwd)
+          GO111MODULE=on go test --failfast --mod=readonly "./test" --linkerd="$HOME/.linkerd2/bin/linkerd" --helm-path="$projectdir/bin/helm" --smi-helm-chart="$projectdir/charts/linkerd-smi" --smi-helm-version="$TAG" --integration-tests
+        else
+          GO111MODULE=on go test --failfast --mod=readonly "./test" --linkerd="$HOME/.linkerd2/bin/linkerd" --integration-tests
+        fi
```
Comment (alpeb) [inline .github/workflows/integration_tests.yml]: I think it looks better if you have two separate steps (guarded with `if`s) with dynamic names so they are visible in the UI `name: Run Integration Tests ({{ $matrix.test }})`

File: charts/linkerd-smi/values.yaml [hunk 2]
```diff
@@ -2,10 +2,9 @@ namespace: linkerd-smi
 
 installNamespace: true
 clusterDomain: cluster.local
-linkerdSMIVersion: &linkerd_smi_version linkerdSMIVersionValue
 
 adaptor:
   image:
     name: cr.l5d.io/linkerd/smi-adaptor
-    version: *linkerd_smi_version
+    version: linkerdSMIVersionValue
```
Comment (alpeb) [inline charts/linkerd-smi/values.yaml:9]: Are you planning on including something like `bin/helm-build` to replace this placeholder and to package the chart, in a separate release PR?

File: test/install_test.go [hunk 3]
```diff
@@ -77,37 +82,104 @@ func TestTrafficSplitsConversionWithSMIAdaptor(t *testing.T) {
...
+		"--set", "identity.issuer.tls.crtPEM=" + helmTLSCerts.Cred.Crt.EncodeCertificatePEM(),
+		"--set", "identity.issuer.tls.keyPEM=" + helmTLSCerts.Cred.EncodePrivateKeyPEM(),
+		"--set", "identity.issuer.crtExpiry=" + helmTLSCerts.Cred.Crt.Certificate.NotAfter.Format(time.RFC3339),
+	}
+
+	if stdout, stderr, err := TestHelper.HelmInstall("linkerd-edge/linkerd2", "linkerd", args...); err != nil {
+		linkerdtestutil.AnnotatedFatalf(t, "'helm install' command failed\n%s\n%s\n%v", stdout, stderr, err)
+	}
+
+	o, err := TestHelper.Kubectl("", "--namespace=linkerd", "rollout", "status", "--timeout=60m", "deploy/linkerd-destination")
```
Comment (alpeb) [inline test/install_test.go]: I think you wanna wait on all deployments, not just destination

**Review by alpeb — COMMENTED** [review]
Looking good :+1: 
A few minor comments below.

File: charts/linkerd-smi/values.yaml (hunk 2 above)
Comment (Pothulapati) [reply charts/linkerd-smi/values.yaml:9]: Yeah, That will be part of the release PR! Didn't add it as it is not needed for the integration tests! :)

**Review by alpeb — APPROVED** [review]
Thanks @Pothulapati , LGTM :+1:

**Review by adleong — APPROVED** [review]

**Pull request merged**